doc/SDK_NOTIFICATIONS.md

```diff
@@ -0,0 +1,77 @@
+- The client initiate a swap and send funds to the on-chain address.
+- The client registers to a push notification for the address using web hook.
+- Once the transaction is confirmed the notification server send a POST request to the associated webhook with the payload that includes the notification type and data.
+- The web hook notifies the delivery services that construct the specific notification format and payload and send it to the device. In that case the push message can be silent, waking up the app to run the signer and complete the swap payment.
+
+## Suggested implementation
+There are several parts for achieving the above:
+
+### SDK
+- `register_notifications(url, platform, token)` API method for the user to register all kind of notifications.
+- Build the webhook URL (defined later in the document) and use it in the following cases:
+ - When creating an on-chain receive swap address - Use breez server API to register for notification of any tx associated with this address
+ - Use LSP API to register webhook to be notified when there is an incoming paypment(htlc) for the user node.
+
+### LSP
+- `subscribe_notifications(url, signature)` API for registering a webhook by sdk clients. The signature is the url signed by the node id. The LSP should maintain a mapping between node ids to web hooks URLs. These webhooks (from the client perspective) tend to change frequently as for example in push notifications a new token is generated from time to time so the client is responsible to update the url on any change.
+
+When an htlc is intercepted in the LSP and the destination peer is not online the LSP will notify the web hook with the right type and data, signling incoming payment, in attempt to wake up the destination node, wait for a some time (how much?) and once the peer is online will forward the htlcs.
+
+### Chain notification service
+Breez server will implement the chain notifications for sdk clients. Connected to bitcoind breez server download every block, check for webhook registration and trigger notifications. API key is needed to subscribe for such notification.
+The following endpoints are needed:
+- `registration_id add_registration(current_registration_id, url)` - get the current registration_id if exists or create one if the former was deleted. Also associate the url with the new registration if is not already associated. At the first time this endpoint is called the `current_registration_id` can be null. This mechanism allows the service to delete expired registrations according to its own policy requiring the client to refresh the registration from time to time signaling it is truly active.
+- `subscribe_address_transactions(registration_id, url, address)` API for subscribing to address notifications.
+- `subscribe_tx_confirmation(registration_id, url, txid)` API for subscribing to specific transaction notifications.
```

Comment:
Very clear and concise overall, thanks.

One question about the chain notification service for the SDK: since the `url` is an argument for subscribing to these notifications, I assume it's not possible to subscribe, for example, in the CLI app, correct?

Reply:
Good question. 
I started this doc by writing that the goal of the SDK is to provide the trigger to the delivery service (push, mail, etc...) So in fact it is the responsibility of the app to connect to the delivery service (e.g subscribe to push notifications) and the app provider to supply the delivery service (e.g a backend service that actually send the push notifications, email, etc..). 
In case we want the cli to receive notifications this kind of design would work as well, for example: We can implement the delivery service as a gRPC service that the cli connects to on startup with some random token and registers for webook that contains that random token.
The hook points to the gRPC service and once triggered, the service identify the right subscriber associated with that token and send a gRPC message over the stream.

Reply:
Saying all the main use case for notifications is to be notified when not online so I don't see a need for that in the cli which by its nature is always online.